packages/language/src/parser/parser-state.ts

```diff
@@ -177,12 +182,17 @@ export class ParserState {
     element: SyntaxNode | undefined,
     kind: CstNodeKind | undefined,
     tokenType: TokenType,
+    code?: SimplePLICode,
```

Comment:
Feels a bit hacky, because we suddenly care about error codes. But I have no good idea for an alternative (only promises came up, like you do an `await consume(...).catch(() => diagnosticFromCode(...))`).

Apart from that, we are excluding ParametricPLICodes (not sure if we need them here really). An alternative could be a lambda as a factory for a diagnostic.

Reply:
In synchronous code you would create a try-catch around it, which make the code bigger/more complex.